src/client.cpp

```diff
@@ -406,7 +406,8 @@ void Client::step(float dtime)
 	// Control local player (0ms)
 	LocalPlayer *player = m_env.getLocalPlayer();
 	assert(player);
-	player->applyControl(dtime, &m_env);
+	player->applyControl(dtime, player->getPlayerControl(),
```

Comment:
why we pass already owned objects to the player class ?

Reply:
It's been a while, but some of these things are because the same code (the same methods) need to be called on the server and the client side, where different things are available.

Or it really is unnecessary, but I'll need to get into the code again first.